akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -424,7 +423,7 @@ private[akka] trait RememberingShard { selfType: Shard ⇒
 
   protected def restartRememberedEntities(): Unit = {
     rememberedEntitiesRecoveryStrategy.recoverEntities(state.entities).foreach { scheduledRecovery ⇒
-      import context.dispatcher
+      implicit val ec: ExecutionContext = context.dispatcher
```

Comment:
Only triggered by messages, a bit tricker to follow, but the `foreach` is on `Set` so this happens when `restartRememberEntities` is called, not later so original is safe to keep.